roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -429,124 +429,123 @@ def get_property_data(self, property_name):
         return ret_val
 
     @staticmethod
-    def write_segmentation(segext_obj, savepath, metadata_dict, **kwargs):
+    def write_segmentation(segext_obj, savepath, metadata_dict=None, **kwargs):
```

Comment:
A few problems here:

1. You have called this variable "metadata_dict", but you allow it to be a string. Why not call it "metadata"? 
2. If you allow this to be multiple types, you should use a docstring to indicate this.
3. You allow this to be `None`, which means it is optional, but there are several places in this function that would break if `metadata_dict` is not provided.

Reply:
I was thinking we make the metadata as an attribute of the class itself. THe user can fill in a yaml file and store the location in the variable: self.metadata = 'path-to-yaml' OR dict 
That way we also wouldn't need this as a required arg. 
What do you think?

Reply:
I'd like to keep this method static so that it maintains the same syntax as all of the other writers across SpikeInterface and RoiExtractors.

Reply:
it is static